sdk/batch/azure-compute-batch/src/test/java/com/azure/compute/batch/SharedKeyTests.java

```diff
@@ -41,59 +41,63 @@ public void testPoolCRUD() {
             updatedMetadata.add(new MetadataItem("foo", "bar"));
-            BatchPoolUpdateParameters poolUpdateParameters = new BatchPoolUpdateParameters(new ArrayList<>(),
-                    new ArrayList<>(),
-                    updatedMetadata);
 
-            poolUpdateParameters.setTargetNodeCommunicationMode(NodeCommunicationMode.SIMPLIFIED);
+            BatchPoolUpdateOptions poolUpdateOptions = new BatchPoolUpdateOptions();
+            poolUpdateOptions.setCertificateReferences(new ArrayList<>())
+                                .setApplicationPackageReferences(new ArrayList<>())
+                                .setMetadata(updatedMetadata);
 
-            poolClientWithSharedKey.updateProperties(sharedKeyPoolId, poolUpdateParameters);
+            poolUpdateOptions.setTargetNodeCommunicationMode(NodeCommunicationMode.SIMPLIFIED);
 
-            pool = poolClientWithSharedKey.get(sharedKeyPoolId);
+            batchClientWithSharedKey.updatePool(sharedKeyPoolId, poolUpdateOptions);
+
+            pool = batchClientWithSharedKey.getPool(sharedKeyPoolId);
             Assertions.assertEquals(NodeCommunicationMode.SIMPLIFIED, pool.getTargetNodeCommunicationMode());
             List<MetadataItem> metadata = pool.getMetadata();
             Assertions.assertTrue(metadata.size() == 1 && metadata.get(0).getName().equals("foo"));
 
             /*
-             * Patch Pool
+             * Update Pool
              */
             updatedMetadata.clear();
             updatedMetadata.add(new MetadataItem("key1", "value1"));
-            BatchPoolPatchParameters poolPatchParameters = new BatchPoolPatchParameters().setMetadata(updatedMetadata).setTargetNodeCommunicationMode(NodeCommunicationMode.CLASSIC);
-            Response<Void> patchPoolResponse = poolClientWithSharedKey.patchWithResponse(sharedKeyPoolId, BinaryData.fromObject(poolPatchParameters), null);
-            HttpRequest patchPoolRequest = patchPoolResponse.getRequest();
-            HttpHeader ocpDateHeader = patchPoolRequest.getHeaders().get(HttpHeaderName.fromString("ocp-date"));
+            BatchPoolUpdateOptions poolUpdateOptions2 = new BatchPoolUpdateOptions().setMetadata(updatedMetadata).setTargetNodeCommunicationMode(NodeCommunicationMode.CLASSIC);
+            Response<Void> updatePoolResponse = batchClientWithSharedKey.updatePoolWithResponse(sharedKeyPoolId, BinaryData.fromObject(poolUpdateOptions2), null);
+            HttpRequest updatePoolRequest = updatePoolResponse.getRequest();
+            HttpHeader ocpDateHeader = updatePoolRequest.getHeaders().get(HttpHeaderName.fromString("ocp-date"));
             Assertions.assertNull(ocpDateHeader);
-            HttpHeader dateHeader = patchPoolRequest.getHeaders().get(HttpHeaderName.DATE);
+            HttpHeader dateHeader = updatePoolRequest.getHeaders().get(HttpHeaderName.DATE);
             Assertions.assertNotNull(dateHeader);
 
```

Comment:
Since this is calling the update Pool API with the raw response, we should also assert against the Authorization Shared Key header :)